.NET/Microsoft.Recognizers.Text.DateTime/Utilities/DateContext.cs

```diff
@@ -20,6 +78,38 @@ public static DateObject SwiftDateObject(DateObject beginDate, DateObject endDat
             return beginDate;
         }
 
+        public (DateTimeParseResult pr1, DateTimeParseResult pr2) SyncDateEntityInFeb29th(DateTimeParseResult pr1, DateTimeParseResult pr2)
```

Comment:
For case: 2.29-3.1 reference: 2019-03-22 
The pastresolution was (2016-2-29, 2019-3-1).
We can synchronize the year with this method.

Reply:
I'm not sure what you mean by this. It seems the past resolution was correctly calculated?

Reply:
I see now. Your resolution to find a leap year for such cases is ok.